File: spark-utils/scala/bible/selectDP2DF.scala
/* use sql context */
import org.apache.spark.sql._
val sqlCtx = new org.apache.spark.sql.hive.HiveContext(sc)
val book_df  = sqlCtx.sql("SELECT * FROM bible.orc_part1_book WHERE book = 'Jn'")
val verse316_df = book_df.filter("cidx = 3").where("vidx = 16")
val verse_df = verse316_df.select("verse")
verse_df.collect().foreach(println)

File: spark-utils/python/bible/selectDP2DF.py
# use sql context 
import sys
from pyspark import SparkContext 
from pyspark.sql import HiveContext, Row 
sc = SparkContext(appName="PySelectDP2")
sqlCtx = HiveContext(sc)
book_df = sqlCtx.sql("SELECT * FROM bible.orc_part1_book WHERE book = 'Jn'")
verse_df = book_df.where("cidx = 3 and vidx = 16").select("verse")
print verse_df.collect()

